ansible_ai_connect/ai/api/tests/test_views.py

```diff
@@ -2647,10 +2683,11 @@ def assert_test(
         with self.assertLogs(logger="root", level="DEBUG") as log:
             r = self.client.post(reverse("explanations"), self.payload, format="json")
             self.assertEqual(r.status_code, expected_status_code)
-            self.assert_error_detail(
-                r, expected_exception().default_code, expected_exception().default_detail
-            )
-            self.assertInLog(expected_log_message, log)
+            if expected_status_code != HTTPStatus.OK:
```

Comment:
Would it not be better to check if `excepted_exception() !== None`?

[This](https://github.com/ansible/ansible-ai-connect-service/blob/c5462a42a7e20e403ff6a07cec82fbf00ef9a6f9/ansible_ai_connect/ai/api/tests/test_views.py#L2849-L2854) example looks like it'd break with your change.